packages/hash/api/src/model/knowledge/account.fields.ts

```diff
@@ -0,0 +1,77 @@
+import { OrgModel, UserModel } from "..";
+import { GraphApi } from "../../graph";
+import {
+  generateSchemaBaseUri,
+  generateWorkspacePropertyTypeSchema,
+  RESTRICTED_SHORTNAMES,
+  workspaceTypesNamespaceUri,
+} from "../util";
+
+module AccountId {
+  // Generate the schema for the account id property type
+  export const accountIdPropertyType = generateWorkspacePropertyTypeSchema({
+    title: "Account ID",
+    possibleValues: [{ primitiveDataType: "Text" }],
+  });
+
+  export const accountIdBaseUri = generateSchemaBaseUri({
+    namespaceUri: workspaceTypesNamespaceUri,
+    kind: "propertyType",
+    title: accountIdPropertyType.title,
+  });
+}
+
+module Shortname {
+  // Generate the schema for the shortname property type
+  export const shortnamePropertyType = generateWorkspacePropertyTypeSchema({
+    title: "Shortname",
+    possibleValues: [{ primitiveDataType: "Text" }],
+  });
+
+  export const shortnameBaseUri = generateSchemaBaseUri({
+    namespaceUri: workspaceTypesNamespaceUri,
+    kind: "propertyType",
+    title: shortnamePropertyType.title,
+  });
+
+  // Validations for shortnames
```

Comment:
Should we add an asana task and TODO to re-express these as datatype constraints?

Reply:
Added in 43df4b08984fa92c95a6a99b83841a7a1baa5d08 referencing https://app.asana.com/0/0/1202900021005257/f